### PR TITLE
Fix/signature verification

### PR DESCRIPTION
**Pull request by oscar-hernandez-nttd**
Change the verification func implementation to address a couple of issues:

- [x] Function not actually working
- [x] Use a method like [secure_compare](https://www.rubydoc.info/gems/rack/Rack%2FUtils:secure_compare) or [crypto.timingSafeEqual](https://nodejs.org/api/crypto.html#cryptotimingsafeequala-b), which performs a "constant time" string comparison to help mitigate certain timing attacks against regular equality operators, or regular loops in JIT-optimized languages.

### CODE REVIEW

File: codeBuildHandler.py [hunk 1]
```diff
@@ -164,45 +191,45 @@ def prepare_response(status_code, detail="An unknown error has occurred."):
...
 
-    return code_build_env_vars
+    return code_pipeline_env_vars
 
 
-def start_codebuild_job(project_name, env_vars: dict):
+def start_codepipeline_job(project_name, env_vars: dict):
 
-    logger.info(f"Starting CodeBuild job for project: {project_name}")
+    logger.info(f"Starting CodePipeline job for project: {project_name}")
```
Comment (aaron-christian-nttd) [inline codeBuildHandler.py]: CodePIpeline is not a job or project. Will also need that var changed to match

File: codeBuildHandler.py [hunk 2]
```diff
@@ -83,30 +107,33 @@ def lambda_handler(event, context):
...
         merged_env_vars['GIT_TOKEN'] = str(git_token)
         merged_env_vars['LATEST_SHORT_HASH'] = merged_env_vars.get('LATEST_COMMIT_HASH', "")[:7]
         merged_env_vars["CODEBUILD_STATUS"] = "INPROGRESS"
-        merged_env_vars["CALLBACK_DESCRIPTION"] = f"CodeBuild job with id: {codebuild_id} is submitted successfully."
+        merged_env_vars["CALLBACK_DESCRIPTION"] = f"CodeBuild job with id: {codepipeline_id} is submitted successfully."
         # Invoke the Git callback and update the build as "INPROGRESS"
         status = invoke_git_callback(merged_env_vars, auth)
         # Respond to the webhook request
-        return prepare_response(status, f"Codebuild stated with an id: {codebuild_id}")
+        return prepare_response(status, f"Codebuild stated with an id: {codepipeline_id}")
```
Comment (aaron-christian-nttd) [inline codeBuildHandler.py]: CodeBuild ref.

File: codeBuildHandler.py [hunk 3]
```diff
@@ -43,10 +43,34 @@ def validate_lambda_env_vars(env_vars: dict):
             valid = False
     return valid, " ".join(validation_errors)
 
+def entrypoint():
+    event = {
```
Comment (aaron-christian-nttd) [inline codeBuildHandler.py]: Is this test data? This seems like it needs to be removed.

Comment (oscar-hernandez-nttd) [reply codeBuildHandler.py]: You are right, let me remove that.

File: codeBuildHandler.py (hunk 1 above)
Comment (oscar-hernandez-nttd) [reply codeBuildHandler.py]: Changed the message for something more descriptive. I believe that solves this conversation.

File: codeBuildHandler.py (hunk 2 above)
Comment (oscar-hernandez-nttd) [reply codeBuildHandler.py]: I believe this was solved in the latest commits

**Comment by oscar-hernandez-nttd** [comment]
Addressing latest change requests and still passing the tests:
Screenshot 2024-10-28 at 5 46 24 p m

**Pull request merged**